src/rgw/librgw.cc

```diff
@@ -513,40 +508,13 @@ namespace rgw {
     init_timer.add_event_after(g_conf()->rgw_init_timeout, new C_InitTimeout);
     mutex.unlock();
 
-    common_init_finish(g_ceph_context);
-
-    rgw_tools_init(this, g_ceph_context);
-
-    rgw_init_resolver();
-    rgw::curl::setup_curl(boost::none);
-    rgw_http_client_init(g_ceph_context);
-
-    auto run_gc =
-      g_conf()->rgw_enable_gc_threads &&
-      g_conf()->rgw_nfs_run_gc_threads;
+    /* stage all front-ends (before common-init-finish) */
+    main.init_frontends1(true /* nfs */);
 
-    auto run_lc =
-      g_conf()->rgw_enable_lc_threads &&
-      g_conf()->rgw_nfs_run_lc_threads;
-
-    auto run_quota =
-      g_conf()->rgw_enable_quota_threads &&
-      g_conf()->rgw_nfs_run_quota_threads;
-
-    auto run_sync =
-      g_conf()->rgw_run_sync_thread &&
-      g_conf()->rgw_nfs_run_sync_thread;
-
-    StoreManager::Config cfg = StoreManager::get_config(false, g_ceph_context);
-    store = StoreManager::get_storage(this, g_ceph_context,
-					 cfg,
-					 run_gc,
-					 run_lc,
-					 run_quota,
-					 run_sync,
-					 g_conf().get_val<bool>("rgw_dynamic_resharding"));
+    common_init_finish(g_ceph_context);
 
-    if (!store) {
+    main.init_storage();
+    if (! main.store) {
```

Comment:
this access to member variable `main.store` seems to be the only reason `RGWLib` needs to be a friend of `AppMain`, can you use `main.get_store()` instead?

Reply:
sure